### ISSUE TITLE
Custom rules?

### ISSUE THREAD

**Issue by oatkiller** [sasstools/sass-lint#1046]
Is it possible to specify user created rules when running sass-lint? I have an idea for a rule. I would like to write it, and try it out on my project for a while before contributing it. I could contribute it straight away, but the idea for my rule could use refinement. 

Ideally, I would be able to do this without forking sass-lint. Thanks!

**Comment by DanPurdy** [comment]
Hopefully soon, its been in our plans for a long time just unfortunately not had the time for a LONG time

**Comment by rootwork** [comment]
I'd also like to see this, not only to try out possible contributions, but for rules that would be project-specific. For instance, I'm working on a project where we're using a Sass map for all `z-index` values, so an actual `z-index: 1`–type declaration should never happen. I'm guessing a "no-z-index" rule would not be that helpful to others, but it'd be nice to be able to create our own for this kind of thing.

**Comment by artemgurzhii** [comment]
@rootwork I think rule definition should start with the custom prefix - `custom` for example) it will be intuitively and syntactically understandable that they are custom rules, and also this will save us from some future rule conflicts, which may appear after upgrading `sass-lint` version. Also, I think that it would be great too have here some of the examples of in which cases and how users are going to use this feature, so this feature will work for them when PR lands.

My use case: We have some mixins, for example mixin `size` with the following code
```sass
@mixin size($width, $height: $width)
  height: $height
  width: $width
```
In short way, set `height` and `width` of the element, `width` comes as the first argument and if `height` is not passed as second argument - use first argument(`width`) as it's value.

So the use case is to disable using of the `height` and `width` properties in the one place, in favor of `include size` mixin.

Rule specification: `custom-no-width-and-height: 1`
Rule description: 
`my-project/config/sass-lint/custom-rules/no-width-and-height.js`
```js
export default {
  throwErrorWhen: 'rules-used-together'
  properties: ['height', 'width'],
  error: {
   message: 'lorem ipsum dolor sit amet' // some custom error message
  }
};
```

when on the lib side

```js
const CUSTOM_POSSIBLE_THROW_ERROR_OPTIONS = {
  'rules-used-together'({ properties, error }, code) {
    const isValid = properties.every(property => code.rules.includes(property));
    if (isValid === false) {
      throw Error(error);
    }
  }
};
```

This is a very abstract version of how it should work. The user needs minimum LoC to describe custom rule.

**P.S.** Is there any ideas/suggestions how to implement this in the best? I would like to help here, as this feature is highly needed for my further work.

**P.P.S** Also, use cases:
Disable use of media queries(except one `_mixins.sass` file), and use a mixin, where all media queries will we described
